Ui web api

zxdcm opened the pull request.

zxdcm committed:
Pagination added.
queryService page bug fixed.

zxdcm committed:
Room size select options now fetches from server

zxdcm committed:
City & Country queries & controllers added.

zxdcm committed:
RoomSizes controller action and query added.

zxdcm committed:
maxRoomSize option added.
CORS allowed

Commit:
Autofac IOC added.

Commit:
Initial pageInfo state updated.

zxdcm committed:
HotelImage(s) queries and controller actions added.

Commit:
HotelList query fixed.

Commit:
Temp pagination added.

Commit:
Styles added

Commit:
markup fixes

Commit:
Markup fixes.
Rest actions added.

Commit:
package updates.
background image uploaded.

zxdcm committed:
Style fixes

zxdcm committed:
Style fixes

zxdcm committed:
queries rewrited. solution a bit cleared

zxdcm committed:
styles splitted in separate files

zxdcm committed:
Commented blockes removed.
Markup fixes

zxdcm committed:
Style fixes

zxdcm committed:
fix

zxdcm committed:
fix

Commit:
Merge branch 'Ui-WebApi' of https://github.com/zxdcm/BookingSystem into Ui-WebApi

Commit:
query restored to "at least working"

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/CountryQueries/Queries/ListCountriesQuery.cs:
```diff
@@ -1,12 +1,50 @@
+                    CountryId = country.CountryId,
+                    CountryName = country.Name
+                })
+                .WhereIf(!string.IsNullOrWhiteSpace(query.CountryNameLike),
```
Have you checked how it is translated to SQL? It may make a subquery to select the necessary columns and then filter the result set. Instead you should apply all filters first and then select columns, that way no subqueries will be created

zxdcm replied:
That was ok but fixed.

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/ImageQueries/Queries/HotelImageQuery.cs:
```diff
@@ -0,0 +1,40 @@
+        {
+            var queryResult = from image in _dataContext.Images
+                join hotelImage in _dataContext.HotelImages on image.ImageId equals hotelImage.ImageId
+                              where hotelImage.HotelId == query.HotelId
```
why is bigger indent made here?

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/ImageQueries/Queries/HotelImageQuery.cs:
```diff
@@ -0,0 +1,40 @@
+
+namespace BookingSystem.Queries.Queries.ImageQueries.Queries
+{
+    public class HotelImageQuery : IQuery<string> 
```
Is it used anywhere? Can't come up with the case.

ssachkovskaya commented on BookingSystem/BookingSystem.WebApi/Utils/PaginationExtension.cs:
```diff
@@ -0,0 +1,27 @@
+
+namespace BookingSystem.WebApi.Utils
+{
+    public static class PaginationExtension
```
you have already introduce such an extension in BookingSystem/BookingSystem.Queries/Infrastructure/PaginationExtension.cs

ssachkovskaya commented on BookingSystem/BookingSystem.WebApi/appsettings.json:
```diff
@@ -7,13 +7,14 @@
   },
   "Jwt": {
-    "Key": "",
+    "Key": "kekekekekkekekkkekekekekekkekekekekkekekekekeek",
```
emmm...

ssachkovskaya commented on BookingSystem/Frontend/src/hotelSearch/components/HotelSearch.js:
```diff
@@ -22,49 +24,62 @@ const HotelSearch = props => {
-              className="spinner-border"
-              style={{ size: "lg", color: "secondary" }}
+              className="spinner-border text-light"
+              style={{ size: "lg", color: "light" }}
```
inline styling is a bad practice

ssachkovskaya commented on BookingSystem/Frontend/src/hotelSearch/containers/HotelSearchContainer.js:
```diff
@@ -28,36 +26,45 @@ class HotelSearchContainer extends Component {
   };
 
   handleReset = event => {
     this.setState({
```
why don't you define default options separately to reuse them here and in the constructor instead of duplicating this code?

ssachkovskaya commented on BookingSystem/Frontend/src/hotelSearch/reducers/hotelSearchReducer.js:
```diff
@@ -21,6 +22,7 @@ const fetchHotelsMap = {
 
 const hotelsInitialState = {
   hotels: [],
+  pageInfo: { page: 1, pageSize: config.pageSize, totalPages: 1 },
```
hm, which rule do you follow to choose between one-line and multi-line initialization?

zxdcm replied:
linter do that

ssachkovskaya commented on BookingSystem/Frontend/src/hotelSearch/containers/HotelSearchContainer.js:
```diff
@@ -28,36 +26,45 @@ class HotelSearchContainer extends Component {
   state = {
```
Thinking if the filters should be a part of the store... It seems to be necessary there to have the complete state: filters, page option and list of hotels. What do you think?

ssachkovskaya commented on BookingSystem/Frontend/src/shared/components/Pagination.js:
```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+
+class Pagination extends Component {
+  getPages(totalPages, currentPage, pageSize) {
```
the common practice is to return a total number of items, but not pages. For example, below you set page size to 10 if it is not specified in arguments, but the totalPages might have been calculated for the different page size

ssachkovskaya commented on BookingSystem/Frontend/src/shared/utils/queryService.js:
```diff
@@ -5,26 +5,27 @@ import moment from "moment";
-    if (data.page) {
-      formattedData.page = data;
-    }
+    // if (data.roomSize) {
```
shall it be removed?

ssachkovskaya commented on BookingSystem/Frontend/src/shared/utils/queryService.js:
```diff
@@ -5,26 +5,27 @@ import moment from "moment";
     "?" +
     Object.keys(data)
-      .filter(key => data[key] != null)
+      .filter(key => data[key] != null && data[key] !== "")
```
.filter(key => data[key]) - will this work?

ssachkovskaya commented on BookingSystem/Frontend/src/styles/card.scss:
```diff
@@ -0,0 +1,11 @@
+    box-sizing: border-box;
+}
+
+.card-img {
```
shall it be inside .card container? If yes => .card { .card-img { } }

ssachkovskaya commented on BookingSystem/Frontend/src/shared/utils/imageService.js:
```diff
@@ -1,9 +1,12 @@
 class ImageService {
   static getHotelImageLink(hotelId) {
-    return links.getHotelImage(hotelId);
+    RequestService.get(links.getHotelImage(hotelId)).then(img => {
```
why do you need then-clause?

zxdcm committed:
Unused code removed.
List countries query fixed.
key & connection string are removed from appsettings

zxdcm committed:
Fixes.
